app/src/main/java/studio/codable/customkeyboard/adapter/CustomKeyboardAdapter.kt

```diff
@@ -21,9 +22,8 @@ class CustomKeyboardAdapter(
                 Glide.with(image.context)
                     .load(animal.imageUrl)
                     .into(image)
-                root.setOnClickListener { onClickItem(animal) }
+                root.setOnClickListener { interactionListener.onClickItem(animal) }
```

Comment:
this is not what I meant, the idea is to have a variable like 

```
val onClickListener = View.OnClickListener { 
         if(item != null){
             onClickitem(item)
         }
    }
```

where animal is like `var animal: Animal? = null` and you set it's value in `onBind` method

then you set your click listener like this: `root.setOnClickListener(onClickListener)`

this way there is only one object of type `View.OnClickListener` per viewholder, instead of a new one being created in each `onBind` invocation